rotary/rules/polish.smk

```diff
@@ -122,61 +120,53 @@ rule polish_polypolish:
-        polca_output = temp("{sample}/polish/polca/{sample}_polca.fasta"),
-        polypolish_sam = temp("{sample}/polish/polca/{sample}_polypolish.fasta.unSorted.sam"),
-        polypolish_bam = temp("{sample}/polish/polca/{sample}_polypolish.fasta.alignSorted.bam"),
-        read_mapping_files= temp(multiext("{sample}/polish/polca/{sample}_polypolish.fasta.bwa",
-            *READ_MAPPING_FILE_EXTENSIONS)),
-        # Add polca directory to output, so it is deleted on rerun. It was causing an error otherwise.
-        polca_directory = directory("{sample}/polish/polca/")
+        polished = "{sample}/polish/pypolca/{sample}_corrected.fasta",
+        report = "{sample}/stats/polish/pypolca_changes.report",
+        variant_calling = temp("{sample}/polish/pypolca/{sample}.vcf"),
+        intermediate_logs = temp(directory("{sample}/polish/pypolca/logs"))
     conda:
-        "../envs/masurca.yaml"
+        "../envs/pypolca.yaml"
     log:
-        "{sample}/logs/polish/polca.log"
+        "{sample}/logs/polish/pypolca.log"
     benchmark:
-        "{sample}/benchmarks/polish/polca.txt"
+        "{sample}/benchmarks/polish/pypolca.txt"
     params:
-        outdir="{sample}/polish/polca"
+        outdir = "{sample}/polish/pypolca",
+        report= "{sample}/polish/pypolca/{sample}.report",
+        careful = "--careful" if str(config.get("careful_short_read_polishing")).lower() == "true" else "",
+        mem_per_thread = int(config.get("memory") / config.get("threads",1))
     threads:
         config.get("threads",1)
     resources:
         mem=config.get("memory")
     shell:
         """
-        printf "### Replace bwa with bwa-mem2 ###\n" >> {log}
-        bwa_path="$(which bwa)"
-        if [ -z "${{bwa_path}}" ]; then
-            bwa_mem2_path="$(which bwa-mem2)"
-            bwa_mem2_dir="$(dirname "${{bwa_mem2_path}}")"
-            ln -s "${{bwa_mem2_path}}" "${{bwa_mem2_dir}}/bwa"
-        
-            for variant in avx avx2 avx512bw sse41 sse42; do
-                bwa_mem2_variant_path="$(which bwa-mem2.${{variant}})"
-                ln -s "${{bwa_mem2_variant_path}}" "${{bwa_mem2_dir}}/bwa.${{variant}}"
-            done
-        fi
+        pypolca run -a {input.polished} -1 {input.qc_short_r1} -2 {input.qc_short_r2} -o {params.outdir} \
+            -p {wildcards.sample} {params.careful} -f -t {threads} -m {params.mem_per_thread}G > {log} 2>&1
         
-        printf "\n\n### Run POLCA ###\n" >> {log}
-        cd {params.outdir}
-        polca.sh -a ../../../{input.polished} -r "../../../{input.qc_short_r1} ../../../{input.qc_short_r2}" -t {threads} -m {resources.mem}G > ../../../{log} 2>&1
-        ln -s "{wildcards.sample}_polypolish.fasta.PolcaCorrected.fa" "{wildcards.sample}_polca.fasta"
-        cd ../../../
+        mv {params.report} {output.report}
         
-        printf "\n\n### Done. ###\n"
+        # Delete the logfile generated by pypolca that has a timestamp on the end - we already save another log file
```

Comment:
I had to add this somewhat long code block to identify and remove a copy of the log file, saved by pypolca, that I wanted to be temp. The log file is saved with the date/time stamp on the end, so it seemed like snakemake was unable to identify it as an `output` (due to the wildcard timestamp).

Reply:
Great!